src/model/neuroevolution/neuroevolution.py

```diff
@@ -16,63 +16,32 @@ class Neuroevolution:
     Genetic algorithm.
     """
 
-    _INDIVIDUALS: int = 20
+    _iteration_counter = 1
+    _INDIVIDUALS: int = 70
     """
     Amount of individuals per generation, minimum 3.
     """
 
-    _GOLDEN_TICKETS: int = 4
-    """
-    Amount of individuals allowed to reproduce for sure.
-    """
-
-    _MAX_PARENTS: int = _INDIVIDUALS // 3
-    """
-    Maximal parents count.
-    """
-
-    @staticmethod
-    def _reproduction_probability(
-        individual: AdultIndividual, bound_adaptation: float
-    ) -> float:
-        """
-        Probability of reproduction for every given individual
-        without golden ticket according to bound adaptation.
-        """
-        return (individual.adaptation / bound_adaptation) ** 3
-
-    _MUTATION_CHANCE: float = 0.70
-    """
-    Mutation chance for every child.
-    """
-
-    _REPRODUCTION_RATE: Tuple[int, int, int] = (1, 1, 1)
-    """
-    Proportional chance for reproduction by respectively:
-    1. swapping single weight,
-    2. swapping single neuron,
-    3. swapping entire layer.
-    """
-
-    _MUTATION_RATE: Tuple[int, int, int, int, int] = (7, 7, 7, 7, 2)
+    _MUTATION_RATE: Tuple[int, int, int, int, int] = (10, 3, 0, 5, 1)
     """
     Proportional chance for mutation by respectively (apply only to children with mutation):
-    1. generate random value for single weight,
-    2. generate random value for single bias,
-    3. change the sign of single weight,
-    4. multiply all weights of single neuron by numbers in range 0.5 to 1.5,
+    1. generate random value for single weight, <-- makes diversity
+    2. generate random value for single bias, <-- makes diversity
+    3. change the sign of single weight, <-- very poor, useless
+    4. multiply all weights of single neuron by numbers in range 0.1 to 10,
+    ^^ maintains individuals value, enhance good individuals
     5  random value for all weights of single neuron.
+    ^^ makes strong diversity, useful in case of stagnancy
     """
 
-    _REPRODUCTION_PROBABILITIES = np.array(_REPRODUCTION_RATE) / sum(_REPRODUCTION_RATE)
     _MUTATION_PROBABILITIES = np.array(_MUTATION_RATE) / sum(_MUTATION_RATE)
 
     def __init__(self, networks: List[NeuralNetwork]) -> None:
         self._new_generation: List[ChildIndividual] = [
-            ChildIndividual(nn) for nn in networks
+            ChildIndividual(nn) for nn in networks[:-1]
         ]
-        self.individuals: List[AdultIndividual] = []
-        self._parents: List[AdultIndividual] = []
+        #  one parent is needed in first iteration
```

Comment:
In the previous implementation there was also a split between parents and children and there was no need for any individual marked as a parent at the very beginning. 
This is an important assumption, because parents are expected to have adaptation already computed. -1 as initial adaptation is some kind of workaround which should not be done. 
What's more important: `with parents` parameter in evolve method decides if parents are being evaluated (run) in the simulation. If `with parents` is `False`, then the parent is not evaluated, so it doesn't have any sensible adaptation (it's -1 - taken out of thin air) which may cause an error (if not now, then maybe in the future). 
We have an architecture created for such cases. It prevents us from assigning some made up adaptations to individuals - that's why we have `AdultIndividual` and `ChildIndividual`. Adult has an adaptation, child does not.

Reply:
Done